float32 pos
float32 speed
